backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -0,0 +1,176 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.FindMatchQuery;
+import com.google.sps.data.Match;
+import com.google.sps.data.Participant;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. */
+@WebServlet("/add-participant")
+public class AddParticipantServlet extends HttpServlet {
+
+  // Datastore Key/Property constants
+  private static final String KEY_PARTICIPANT = "Participant";
+  private static final String KEY_MATCH = "Match";
+  private static final String PROPERTY_USERNAME = "username";
+  private static final String PROPERTY_STARTTIMEAVAILABLE = "startTimeAvailable";
+  private static final String PROPERTY_ENDTIMEAVAILABLE = "endTimeAvailable";
+  private static final String PROPERTY_DURATION = "duration";
+  private static final String PROPERTY_TIMESTAMP = "timestamp";
+  private static final String PROPERTY_FIRSTPARTICIPANT = "firstParticipant";
+  private static final String PROPERTY_SECONDPARTICIPANT = "secondParticipant";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Request parameter values
+    // TODO: Check input values are filled (before allowing to submit)
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+    if (email == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid email.");
+      return;
+    }
+    String username = email.split("@")[0];
+
+    String timezone = request.getParameter("timezone");
+    ZoneId zoneId = ZoneId.of(timezone); // TODO: convert input timezone to valid ZoneId
+    ZonedDateTime startTimeAvailable =
+        ZonedDateTime.now(zoneId); // TODO: set to future time if not available now
+    Instant endTimeAvailableInstant =
+        Instant.ofEpochMilli(
+            Long.parseLong(
+                request.getParameter("endTimeAvailable"))); // TODO: figure out input format
+    LocalDateTime endTimeAvailableLocal =
+        endTimeAvailableInstant.atZone(ZoneId.of("UTC")).toLocalDateTime();
+    ZonedDateTime endTimeAvailable = endTimeAvailableLocal.atZone(zoneId);
```

Comment:
Would this work?

`ZoneDateTime endTimeAvailable = endTimeAvailableInstant.atZone(zoneId)`